pkg/reconciler/reconciler.go

```diff
@@ -521,8 +521,12 @@ func (r *Reconciler) Reconcile(ctx context.Context, req ctrl.Request) (res ctrl.
 		return ctrl.Result{}, err
 	}
 
+	shouldUpdate := true
```

Comment:
Introducing a new variable here that is ready by the defer closure (and not easy to grok straight away), I think it might be cleaner to introduce a method on the updater such as `SkipAllUpdates()` or `DoNotApply()` that would:
- cause `Apply(...)` to just `return nil` without doing anything, and
- (optional) cause the updater to `panic()` on further `Ensure...` calls to make sure that `SkipUpdates` is closely followed by a `return` from `Reconcile()`

Reply:
I like this a lot 💯